vowpalwabbit/csoaa.h

```diff
@@ -11,6 +11,25 @@ namespace CSOAA {
   LEARNER::base_learner* csldf_setup(arguments& arg);
   struct csoaa;
   void finish_example(vw& all, csoaa&, example& ec);
+
+  inline example* remove_empty_last_example(multi_ex& ec_seq)
+  {
+    if (ec_seq.size() > 1 && example_is_newline(*ec_seq.last()))
+    {
+      return ec_seq.pop();
+    }
+    return nullptr;
+  }
```

Comment:
It seems like we should not need this because it's double punctuation to have both a vector and a newline example.